### PR TITLE
Upgrade to check-spelling v0.0.22

### PR DESCRIPTION
**Pull request by jsoref**
## What's changed?

* I'm refreshing the metadata using updates to https://github.com/check-spelling/spell-check-this
  * a grammar/spelling fix (caught by the refresh)
  * the workflow will now run when (if) dependabot creates a PR (dependabot is "special")
* I'm dropping a number of dictionaries from the workflow. One (fsharp) happened to have a typo in it (which i've fixed upstream) -- there's a tension between too many dictionaries (with a risk of them also having typos) and too few (which means too many items in expect and thus it's too hard to spot new typos) -- at times I actually drop extra dictionaries from check-spelling entirely (the c++ and win32 dictionaries turned out to be sufficiently bad that suggesting them was a bad idea)
  * a spelling fix (see above)
* converting some `expect` items to `patterns` so that normal typos for the words aren't accepted

check-spelling is smarter :)

## What's your motivation?

Upgrading clients to [v0.0.22](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22)

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [ ] I've added the license header to any new files through `./gradlew licenseFormat`
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: .github/actions/spelling/allow.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
+Controlflow
```
Comment (jsoref) [inline .github/actions/spelling/allow.txt:1]: I'm sorting this file -- check-spelling doesn't care

File: .github/actions/spelling/allow.txt [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
 github
 https
 ssh
 ubuntu
 workarounds
-Controlflow
-controflow
```
Comment (jsoref) [inline .github/actions/spelling/allow.txt:7]: This appears to be a typo that didn't seem necessary

File: .github/actions/spelling/candidate.patterns [hunk 3]
```diff
@@ -156,7 +161,7 @@ GHSA(?:-[0-9a-z]{4}){3}
 # GitLab commits
 \bgitlab\.[^/\s"]*/(?:[^/\s"]+/){2}commits?/[0-9a-f]+\b
 
-# binanace
+# binance
```
Comment (jsoref) [inline .github/actions/spelling/candidate.patterns:164]: Yeah, I had a typo in my files -- the hazard of excluding them by default -- it took a while for me to catch 😊

File: .github/actions/spelling/expect.txt [hunk 4]
```diff
@@ -138,7 +142,6 @@ MDBs
 methoddeclaration
 micronaut
 Migratetohypersistenceutilshibernate
-Migratetohypersistenceutilshibernate6
```
Comment (jsoref) [inline .github/actions/spelling/expect.txt:141]: check-spelling will now complain about things like this (expect only wants words, and by default numbers can't be in words -- although with this version you could actually change the definition of a word...)

File: .github/actions/spelling/line_forbidden.patterns [hunk 5]
```diff
@@ -11,21 +11,27 @@
 # s.b. anymore
 \bany more[,.]
 
+# s.b. cannot
+\b[Cc]an not\b
```
Comment (jsoref) [inline .github/actions/spelling/line_forbidden.patterns:15]: This caught an item below

File: .github/actions/spelling/patterns.txt [hunk 6]
```diff
@@ -1,37 +1,26 @@
...
-# Compiler flags
-(?:^|[\t ,>"'`=(])-[DPWXYLlf](?=[A-Z]{2,}|[A-Z][a-z]|[a-z]{2,})
+# hit-count: 27 file-count: 8
+# Compiler flags (Unix, Java/Scala)
+# Use if you have things like `-Pdocker` and want to treat them as `docker`
+(?:^|[\t ,>"'`=(])-(?:(?:J-|)[DPWXY]|[Llf])(?=[A-Z]{2,}|[A-Z][a-z]|[a-z]{2,})
 
-# hit-count: 202 file-count: 201
-# Non-English
-[a-zA-Z]*[ÀÁÂÃÄÅÆČÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæčçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź][a-zA-Z]{3}[a-zA-ZÀÁÂÃÄÅÆČÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæčçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź]*|[a-zA-Z]{3,}[ÀÁÂÃÄÅÆČÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæčçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź]|[ÀÁÂÃÄÅÆČÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæčçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź][a-zA-Z]{3,}
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:14]: check-spelling didn't suggest this, so apparently whichever content used to be present is no longer checked/matched and thus it didn't make sense to carry it. If it reappears, check-spelling will suggest it again.

File: .github/actions/spelling/patterns.txt [hunk 7]
```diff
@@ -69,3 +61,12 @@
...
 [Ii]s[Pp]armeters[Pp]rovided
+
+# Due to misspelling in a recipe
+JavaxPeristenceXmlToJakartaPersistenceXml
+
+# Due to misspelling in a recipe
+MigrateHibernateContraintsToJavax
+
+# brand
+\[Spring Initializr\]
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:72]: These are to remove a couple of items from expect as noted in the PR description

File: .github/workflows/spelling.yml [hunk 8]
```diff
@@ -93,33 +93,25 @@ jobs:
...
         check_file_names: 1
         spell_check_this: openrewrite/rewrite-docs@master
         post_comment: 0
         use_magic_file: 1
+        warnings: bad-regex,binary-file,deprecated-feature,ignored-expect-variant,large-file,limited-references,no-newline-at-eof,noisy-file,non-alpha-in-dictionary,token-is-substring,unexpected-line-ending,whitespace-in-dictionary,minified-file,unsupported-configuration,no-files-to-check
         experimental_apply_changes_via_bot: ${{ github.repository_owner != 'openrewrite' && 1 }}
         use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
         extra_dictionary_limit: 10
-        dictionary_source_prefixes: |
-          {"cspell": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220816/dictionaries/", "cspell1": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20230509/dictionaries/"}
```
Comment (jsoref) [inline .github/workflows/spelling.yml:108]: the paths used originally were based on the version here, so there's no longer a need for the `1` prefix.

File: .github/workflows/spelling.yml [hunk 9]
```diff
@@ -93,33 +93,26 @@ jobs:
...
-          cspell1:npm/dict/npm.txt
-          cspell1:java/src/java.txt
-          cspell1:k8s/dict/k8s.txt
-          cspell1:typescript/dict/typescript.txt
-          cspell1:filetypes/filetypes.txt
-          cspell1:fullstack/dict/fullstack.txt
-          cspell1:python/src/python/python-lib.txt
-          cspell1:scala/dict/scala.txt
-          cspell1:php/dict/php.txt
-          cspell1:node/dict/node.txt
```
Comment (jsoref) [inline .github/workflows/spelling.yml:122]: a bunch of these didn't seem to have enough value to retain ...

**Review by mike-solomon — APPROVED** [review]
Thanks so much for doing all of this! It really helps :D

**Pull request merged**